optuna/logging.py

```diff
@@ -28,15 +30,16 @@
 _default_handler: Optional[logging.Handler] = None
 
 
-def create_default_formatter() -> colorlog.ColoredFormatter:
+def create_default_formatter() -> Union[Formatter, colorlog.ColoredFormatter]:
     """Create a default formatter of log messages.
 
     This function is not supposed to be directly accessed by library users.
     """
-    return colorlog.ColoredFormatter(
-        "%(log_color)s[%(levelname)1.1s %(asctime)s]%(reset)s %(message)s",
-        no_color=False if _color_supported() else True,
-    )
+    if _color_supported():
+        return colorlog.ColoredFormatter(
+            "%(log_color)s[%(levelname)1.1s %(asctime)s]%(reset)s %(message)s",
+        )
+    return Formatter("[%(levelname)1.1s %(asctime)s] %(message)s")
```

Comment:
I'm not sure, but do you think we can simplify the format strings by extracting the common parts?

```suggestion
    header = "[%(levelname)1.1s %(asctime)s]"
    message = "%(message)s"
    if _color_supported():
        return colorlog.ColoredFormatter(
            f"%(log_color)s{header}%(reset)s {message}",
        )
    return Formatter(f"{header} {message}")
```